File: hw/toapanta/u3/hw35CrudStrategy/CrudStrategyPY/model/strategy/crud_strategy.py
from abc import ABC, abstractmethod
from typing import List
from model.customer import Customer

class CrudStrategy(ABC):
    @abstractmethod
    def add(self, customer: Customer) -> bool:
        pass
    
    @abstractmethod
    def delete(self, customer_id: int) -> bool:
        pass
    
    @abstractmethod
    def update(self, customer_id: int, customer: Customer) -> bool:
        pass
    
    @abstractmethod
    def read_all(self) -> List[Customer]:
        pass
    
    @abstractmethod
    def read_by_id(self, customer_id: int) -> Customer:
        pass
    
    @abstractmethod
    def get_format_name(self) -> str:
        pass

File: hw/aymacana/u3/hw35CrudStrategy/CrudStrategyJS/models/strategies/CrudStrategy.js
export class CrudStrategy {
    async add(customer) {
        throw new Error("Method 'add()' must be implemented");
    }

    async delete(customerId) {
        throw new Error("Method 'delete()' must be implemented");
    }

    async update(customerId, customer) {
        throw new Error("Method 'update()' must be implemented");
    }

    async readAll() {
        throw new Error("Method 'readAll()' must be implemented");
    }

    async readById(customerId) {
        throw new Error("Method 'readById()' must be implemented");
    }

    getFormatName() {
        throw new Error("Method 'getFormatName()' must be implemented");
    }
}